core-container/entry.sh

```diff
@@ -242,51 +220,67 @@ function run_kcp() {
 
 function run_init() {
     echo "--< Starting init >--"
-    wait_kcp_ready
-    set_provider_adapters
-    kubestellar init --local-kcp false --ensure-imw $ENSURE_IMW --ensure-wmw $ENSURE_WMW
-    kubectl ws root
+    create_spaceprovider_object
+    kubestellar init --ensure-imw $ENSURE_IMW --ensure-wmw $ENSURE_WMW
     touch ready
     echo "***READY***"
     sleep infinity
 }
 
-
 function run_mailbox_controller() {
     echo "--< Starting mailbox-controller >--"
     wait-kubestellar-ready
+    # TODO: remove the kcp dependency in PR3
+    get_kcp_kubeconfig
+    KUBECONFIG=$kcp_kubeconfig
     kubectl ws root:espw
     if ! mailbox-controller -v=${VERBOSITY} ; then
         echoerr "unable to start mailbox-controller!"
         exit 1
     fi
 }
 
-
 function run_where_resolver() {
     echo "--< Starting where-resolver >--"
     wait-kubestellar-ready
+    # TODO: remove the kcp dependency in PR3
+    get_kcp_kubeconfig
+    KUBECONFIG=$kcp_kubeconfig
     kubectl ws root:espw
     if ! kubestellar-where-resolver -v ${VERBOSITY} ; then
         echoerr "unable to start kubestellar-where-resolver!"
         exit 1
     fi
 }
 
-
 function run_placement_translator() {
     echo "--< Starting placement-translator >--"
     wait-kubestellar-ready
+    # TODO: remove the kcp dependency in PR3
+    get_kcp_kubeconfig
+    KUBECONFIG=$kcp_kubeconfig
     kubectl ws root:espw
     if ! placement-translator --allclusters-context  "system:admin" -v=${VERBOSITY} ; then
         echoerr "unable to start mailbox-controller!"
         exit 1
     fi
 }
 
+# If a space manager config is not passed into entry.sh then we assume that the host cluster is the space manager core cluster. 
+function set_kubestellar_kubeconfig() {
+    kubectl --kubeconfig $kubestellar_kubeconfig config set-cluster sm-mgt --server="https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT}" --certificate-authority=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+    kubectl --kubeconfig $kubestellar_kubeconfig config set-credentials sm-mgt --token="$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)"
+    kubectl --kubeconfig $kubestellar_kubeconfig config set-context sm-mgt --cluster=sm-mgt --user=sm-mgt
+    kubectl --kubeconfig $kubestellar_kubeconfig config use-context sm-mgt
```

Comment:
What do the "s" and "m" at the start of "sm-mgt" stand for?

Reply:
Consulting with the team, we will change the the name to sm-core in #1286.